crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2038,6 +2145,10 @@ impl<'db> Class<'db> {
             );
         }
 
+        if name == "__class__" {
+            return self.metaclass(db).unwrap_or(Type::Unknown).into();
```

Comment:
The fact that we have to do this in multiple places makes me think that maybe we should have a `try_metaclass` method (that returns a `Result`) and a `metaclass` method (that has a fallback metaclass type), similar to the way we have a `try_mro` method and an `mro` method. That way if we change the fallback in the future (which I think is quite likely -- see my comment above), we'll only have to change it in one place.

Reply:
I'm not really convinced of the ergonomics of that -- I don't think we should hide the need to handle errors from callers -- but I'm fine to do it for consistency.

Reply:
The thing is that in this case, there's really nothing a caller can or should do about the error -- we've already emitted a diagnostic for it in `check_class_definitions`, and once we've done that, the error _shouldn't_ be explicitly handled by any other caller, it would just result in duplicate diagnostics in some form. So it's more _correct_, not just more ergonomic, to provide most callers with an API that just tells them the type of the metaclass is `Unknown`.